### PR TITLE
Change validation functions to take in model.Config object

### PR DESCRIPTION
**Pull request by howardjohn**
This is a dependency for https://github.com/istio/istio/issues/26889 and for future validation we want to add that takes into account object metadata (labels, annotations). Technically we could just add more parameters, but this feels much cleaner.

As a result of this change, an import loop is created. Instead of having the core model.Config object living in pilot, it ought to be in pkg/config, which avoids this loop (at the cost of a massive PR....)

### CODE REVIEW

**Review by rshriram — APPROVED** [review]

File: galley/pkg/config/analysis/analyzers/schema/validation_test.go [hunk 1]
```diff
@@ -35,10 +35,10 @@ func TestCorrectArgs(t *testing.T) {
 
 	m1 := &v1alpha3.VirtualService{}
 
-	testSchema := schemaWithValidateFn(func(name, ns string, msg proto.Message) (errs error) {
-		g.Expect(name).To(Equal("name"))
-		g.Expect(ns).To(Equal("ns"))
-		g.Expect(msg).To(Equal(m1))
+	testSchema := schemaWithValidateFn(func(cfg config.Config) (errs error) {
```
Comment (nmittler) [inline galley/pkg/config/analysis/analyzers/schema/validation_test.go:38]: Is there any way we could just get rid of `Config` and `Meta` and just use the [resource](https://github.com/istio/istio/blob/master/pkg/config/resource/) package?

**Review by nmittler — APPROVED** [review]

File: galley/pkg/config/analysis/analyzers/schema/validation_test.go (hunk 1 above)
Comment (howardjohn) [reply galley/pkg/config/analysis/analyzers/schema/validation_test.go:38]: Discussed offline, we probably can at some point but this PR is large enough

**Review by JimmyCYJ — APPROVED** [review]

**Pull request merged**